_includes/v2.2/app/sqlalchemy-basic-sample.py

```diff
@@ -1,38 +1,108 @@
 class Account(Base):
     __tablename__ = 'accounts'
     id = Column(Integer, primary_key=True)
     balance = Column(Integer)
 
-# Create an engine to communicate with the database. The "cockroachdb://" prefix
-# for the engine URL indicates that we are connecting to CockroachDB.
-engine = create_engine('cockroachdb://maxroach@localhost:26257/bank',
-                       connect_args = {
-                           'sslmode' : 'require',
-                           'sslrootcert': 'certs/ca.crt',
-                           'sslkey':'certs/client.maxroach.key',
-                           'sslcert':'certs/client.maxroach.crt'
-                       })
-Session = sessionmaker(bind=engine)
+
+# Create an engine to communicate with the database. The
+# "cockroachdb://" prefix for the engine URL indicates that we are
+# connecting to CockroachDB using the 'cockroachdb' dialect.
+# For more information, see
+# https://github.com/cockroachdb/cockroachdb-python.
+
+secure_cluster = True           # Set to False for insecure clusters
+connect_args = {}
+
+if secure_cluster:
+    connect_args = {
+        'sslmode': 'require',
+        'sslrootcert': 'certs/ca.crt',
+        'sslkey': 'certs/client.maxroach.key',
+        'sslcert': 'certs/client.maxroach.crt'
+    }
+else:
+    connect_args = {'sslmode': 'disable'}
+
+engine = create_engine(
+    'cockroachdb://maxroach@localhost:26257/bank',
+    connect_args=connect_args,
+    echo=True                   # Log SQL queries to stdout
+)
 
 # Automatically create the "accounts" table based on the Account class.
 Base.metadata.create_all(engine)
 
-# Insert two rows into the "accounts" table.
-session = Session()
-session.add_all([
-    Account(id=1, balance=1000),
-    Account(id=2, balance=250),
-])
-session.commit()
-
-# Print out the balances.
-for account in session.query(Account):
-    print(account.id, account.balance)
+
+# The code below generates random IDs for new accounts.  Since this is
+# just a demo, it does not do any work to ensure the new IDs don't
+# collide with existing IDs.
+
+def create_random_accounts(sess, n):
+    '''Create N new accounts with random IDs and random account balances.
+
+    Note that since this is a demo, we don't do any work to ensure the
+    new IDs don't collide with existing IDs.
+    '''
+    new_accounts = []
+    elems = iter(range(n))
+    for i in elems:
+        billion = 1000000000
+        new_id = floor(random.random()*billion)
+        new_accounts.append(
+            Account(
+                id=new_id,
+                balance=floor(random.random()*1000000)
+            )
+        )
+    sess.add_all(new_accounts)
+
+
+run_transaction(sessionmaker(bind=engine),
+                lambda s: create_random_accounts(s, 100))
+
+
+# Helper for getting random existing account IDs.
+
+def get_random_account_id():
+    res = engine.execute('SELECT id FROM accounts ORDER BY random() LIMIT 1;')
```

Comment:
`ORDER BY random()` is expensive. I think you still want a local set for selecting accounts here. Create a set of all the account IDs created in `create_random_accounts`, just don't worry about filling it with any pre-existing ones.

Reply:
That makes sense.  In the latest update I used a Python set to hold just the most recently created account IDs, and when it's time to shuffle money around we use only those (rather than selecting at random from the entire table).